ci-automation/vendor-testing/equinix_metal.sh

```diff
@@ -0,0 +1,108 @@
+EQUINIXMETAL_INSTANCE_TYPE="${!EQUINIXMETAL_INSTANCE_TYPE_VAR}"
+MORE_INSTANCE_TYPES_VAR="EQUINIXMETAL_${arch}_MORE_INSTANCE_TYPES"
+MORE_INSTANCE_TYPES=( ${!MORE_INSTANCE_TYPES_VAR} )
+
+# We override `EQUINIXMETAL_PARALLEL`, because running kola with EQUINIXMETAL_PARALLEL >= 4 causes the
+# tests to provision >= 12 ARM servers at the same time. As the DA metro does not
+# have that many free ARM servers, the whole tests will fail.
+# In addition, we override `timeout` to 10 hours, because it takes more than 8 hours
+# to run all tests only with 2 tests in parallel.
+if [[ "${arch}" == "arm64" ]]; then
+  EQUINIXMETAL_PARALLEL="2"
+  timeout=15h
+else
+  timeout=8h
+fi
+
+BASE_URL="http://${BUILDCACHE_SERVER}/images/${arch}/${vernum}"
+
+# GCP_JSON_KEY is provided base64 encoded.
+echo "${GCP_JSON_KEY}" | base64 --decode > ./gcp-key.json
+
+run_equinix_metal_kola_test() {
+    local instance_type="${1}"
+    local instance_tapfile="${2}"
+
+    timeout --signal=SIGQUIT "${timeout}" \
+        kola run \
+          --board="${arch}-usr" \
+          --basename="ci-${vernum/+/-}" \
+          --platform=equinixmetal \
+          --tapfile="${instance_tapfile}" \
+          --parallel="${EQUINIXMETAL_PARALLEL}" \
+          --torcx-manifest=../torcx_manifest.json \
+          --equinixmetal-image-url="${BASE_URL}/${EQUINIXMETAL_IMAGE_NAME}" \
+          --equinixmetal-installer-image-kernel-url="${BASE_URL}/${PXE_KERNEL_NAME}" \
+          --equinixmetal-installer-image-cpio-url="${BASE_URL}/${PXE_IMAGE_NAME}" \
+          --equinixmetal-metro="${equinixmetal_metro}" \
+          --equinixmetal-plan="${instance_type}" \
+          --equinixmetal-project="${EQUINIXMETAL_PROJECT}" \
+          --equinixmetal-storage-url="${EQUINIXMETAL_STORAGE_URL}" \
+          --gce-json-key="./gcp-key.json" \
+          --equinixmetal-api-key="${EQUINIXMETAL_KEY}" \
+          "${@}"
+
+    # compare the tested instance with the default instance type.
+    if [[ "${instance_type}" != "${EQUINIXMETAL_INSTANCE_TYPE}" ]]; then
+        sed --in-place "s/cl\.internet/${instance_type}\.cl\.internet" "${instance_tapfile}"
+    fi
+}
+
+cl_internet_included="$(kola list --platform=equinixmetal --filter "${@}" | { grep cl.internet || : ; } )"
+run_more_tests=0
+
+[[ -n "${cl_internet_included}" ]] && [[ "${#MORE_INSTANCE_TYPES[@]}" -gt 0 ]] && run_more_tests=1
+
+if [[ "${run_more_tests}" -eq 1 ]]; then
+    for instance_type in "${MORE_INSTANCE_TYPES[@]}"; do
+	(
+            OUTPUT=$(run_equinix_metal_kola_test "${instance_type}" "validate_${instance_type}.tap" 'cl.internet' 2>&1 || :)
+            printf "=== START $INSTANCE ===\n%s\n=== END $INSTANCE ===\n" "$(sed -e 's/^/prefix: /g' <<<"${foo}")"
+        ) &
+    done
+fi
+
+set -x
+
+run_equinix_metal_kola_test "${EQUINIXMETAL_INSTANCE_TYPE}" "${tapfile}" "${@}"
+
+set +x
+
+if [[ "${run_more_tests}" -eq 1 ]]; then
+    wait
+    cat validate_*.tap >>"${tapfile}"
+fi
+
+rm --force ./gcp-key.json
```

Comment:
The other tests use `trap 'cmd' EXIT` to set cleanup actions up, before kola is invoked

Reply:
or see other suggestion of using a temporary FD with `<(cmd)`